### ISSUE TITLE
Calling passive authentication middleware from MVC/Web API

### ISSUE THREAD

**Issue by BadriL** [aspnet/Mvc#1601]
It is not clear how to call an authentication middleware configured to run in passive mode from MVC/Web API (or whatever). In the previous version, we used to have an authentication filter, specifically `HostAuthenticationFilter` that can do this out of box. In vNext, what is the recommended extension point to call `AuthenticateAsync` on the context, when an authentication filter is not available?

#### Option 1: Use authorization filter
I don't think we can use an action filter because authorization filters run before action filters. But we can use an authorization filter to do this. One good thing is that  we can specify the order of the filter. Hence, the authz filter calling `AuthenticateAsync` on the context can be made to run before other authorization filters. But this does not sound nice because we are re-purposing an authorization filter for authentication. This does not sound nice especially given the fact the earlier version of ASP.NET has a specific filter exactly for this case.

#### Option 2: Don't bother about active/passive mode
Let all the middleware components in the pipeline all run and establish identities. During authorization, be selective and look for a specific authentication type. In fact, the issue [432] (https://github.com/aspnet/Mvc/issues/432) is about introducing authentication type on authorization attribute for this purpose? If that is the case, will you remove the active/passive modes from authentication infrastructure [here] (https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Security/Infrastructure/AuthenticationHandler.cs#L75)?

**Comment by HaoK** [comment]
So the basic idea is, if your authorization policy wants a specific authenticationType, it can ask for it via AuthorizationPolicy.ActiveAuthenticationTypes, if any are specified, the default authorization service will call `httpContext.AuthenticateAsync(ActiveAuthenticationTypes)` and use all of those as the ClaimsPrincipal for the authorization call

**Comment by HaoK** [comment]
At least this is the current state of AuthZ, we are in the early stages of iterating on this, so its still subject to change

**Comment by HaoK** [comment]
Active/Passive have been reworked a bit, and removed from base AuthenticationHandler and pushed to a new derived AutomaticAuthenticationHandler (which today only Cookies and Bearer extend).  Social auth middlewares no longer have this concept

**Comment by ikhanal** [comment]
Can somebody give me a sample code or a guidance on  how to implement **HostAuthenticationFilter** in MVC4/WebApi 2 project where I have implemented Cookie based authentication in MVC4 project and then I added an Area called Api where I have exposed EDM using WebApi2 and OData V3 but I need to implement OAuth 2 token i.e. bearer based authentication in order to access api controller. In addition to this, I need login controller in my api as clients are Native Mobile Apps (iOS and Android users).
